Update README for aggregation and cleanup V5 (#233)
DataDog/datadog-go: README.md

After the client is created, you can start sending Service Checks to Datadog. See the dedicated [Service Check Submission: DogStatsD documentation](https://docs.datadoghq.com/developers/service_checks/dogstatsd_service_checks_submission/?code-lang=go) to see how to submit a Service Check to Datadog.

## Client side aggregation

Starting with version `5.0.0` (and `3.6.0` in beta), the client offers aggregation or value packing on the client side.

This feature aims at reducing both the number of packets sent to the Agent and the packet drops in very high throughput
scenarios.

> The aggregation window is 2s by default and can be changed through `WithAggregationInterval()` options. Note that the

Reviewer: ```suggestion
The aggregation window is 2s by default and can be changed through `WithAggregationInterval()` option. Note that the
```

> aggregation window on the Agent side is 10s for DogStatsD metrics. So for example, setting a aggregation window of 3s on

Reviewer: ```suggestion
aggregation window on the Agent side is 10s for DogStatsD metrics. So for example, setting an aggregation window of 3s in
```

the client will produce a spike in your dashboard every 30 second for counts metrics (as the third 10s bucket on the
Agent will receive 4 samples from the client).

Aggregation can be disabled using the `WithoutClientSideAggregation()` option.

The telemetry `datadog.dogstatsd.client.metrics` is unchanged and represents the number of metrics before aggregation.
New metrics `datadog.dogstatsd.client.aggregated_context` and `datadog.dogstatsd.client.aggregated_context_by_type` have
been introduced. See the [Monitoring this client](#monitoring-this-client) section.

### "Basic" aggregation

Enabled by default, the client will aggregate `gauge, `count` and `set`.

### "Extended" aggregation

> This feature is only compatible with Agent's version >=6.25.0 && <7.0.0 or Agent's versions >=7.25.0.

Reviewer: ```suggestion
This feature is only compatible with Agent versions starting from 6.25.0 and 7.25.0.
```

Author: Actually this is the wording we find with documentation since 7.24 is not compatible for example.

Disabled by default, the client can also pack multiple values for `histogram`, `distribution` and `timing` in one
message. Real aggregation is not possible for those types since the Agent also aggregates and two aggregation levels
would change the final value sent to Datadog.

> When this option is enabled, the agent will buffer the metrics by context (unique combination of metrics name and tag
> set), and flush them in the fewest number of messages.

Reviewer: ```suggestion
When this option is enabled, the agent will buffer the metrics by combination of metric name and tags, and send them in the fewest number of messages.
```
